packages/eas-cli/src/update/configure.ts

```diff
@@ -0,0 +1,291 @@
+      'https://expo.fyi/eas-update-config.md'
+    )}\n`
+  );
+  Log.log(chalk.bold(JSON.stringify(modifyConfig, null, 2)));
+  Log.addNewLineIfNone();
+
+  if (workflows.android === Workflow.GENERIC || workflows.ios === Workflow.GENERIC) {
+    Log.warn(
+      `You will also have to manually edit the projects ${chalk.bold(
+        'Expo.plist/AndroidManifest.xml'
+      )}. ${learnMore('https://expo.fyi/eas-update-config.md#native-configuration')}`
+    );
+  }
+
+  Log.addNewLineIfNone();
+}
+
+/**
+ * Make sure that the current `app.json` configuration for EAS Updates is set natively.
+ */
+export async function ensureEASUpdatesIsConfiguredNativelyAsync(
+  graphqlClient: ExpoGraphqlClient,
+  {
+    exp,
+    projectId,
+    projectDir,
+    platform,
+    workflows,
+  }: {
+    exp: ExpoConfig;
+    projectId: string;
+    projectDir: string;
+    platform: RequestedPlatform;
+    workflows: Record<Platform, Workflow>;
+  }
+): Promise<void> {
+  if (['all', 'android'].includes(platform) && workflows.android === Workflow.GENERIC) {
+    await syncAndroidUpdatesConfigurationAsync(graphqlClient, projectDir, exp, projectId);
+    Log.withTick(`Configured ${chalk.bold('AndroidManifest.xml')} for EAS Update`);
+  }
+
+  if (['all', 'ios'].includes(platform) && workflows.ios === Workflow.GENERIC) {
+    await syncIosUpdatesConfigurationAsync(graphqlClient, projectDir, exp, projectId);
+    Log.withTick(`Configured ${chalk.bold('Expo.plist')} for EAS Update`);
+  }
+}
+
+/**
+ * Make sure EAS Update is fully configured in the current project.
+ * This goes over a checklist and performs the following checks or changes:
+ *   - Enure the `expo-updates` package is currently installed.
+ *   - Ensure `app.json` is configured for EAS Updates
+ *     - Sets `runtimeVersion` if not set
+ *     - Sets `updates.url` if not set
+ *   - Ensure latest changes are reflected in the native config, if any
+ */
+export async function ensureEASUpdatesIsConfiguredAsync(
+  graphqlClient: ExpoGraphqlClient,
+  {
+    exp: expWithoutUpdates,
+    projectId,
+    projectDir,
+    platform,
+  }: {
+    exp: ExpoConfig;
+    projectId: string;
+    projectDir: string;
+    platform: RequestedPlatform;
+  }
+): Promise<{ projectChanged: boolean }> {
+  const hasExpoUpdates = isExpoUpdatesInstalledOrAvailable(
+    projectDir,
+    expWithoutUpdates.sdkVersion
+  );
+  if (!hasExpoUpdates) {
+    await installExpoUpdatesAsync(projectDir, { silent: true });
+    Log.withTick('Installed expo updates');
+  }
+
+  const workflows = await resolveWorkflowPerPlatformAsync(projectDir);
+  const { projectChanged, exp: expWithUpdates } =
+    await ensureEASUpdatesIsConfiguredInExpoConfigAsync({
+      exp: expWithoutUpdates,
+      projectDir,
+      projectId,
+      platform,
+      workflows,
+    });
+
+  if (projectChanged || !hasExpoUpdates) {
```

Comment:
is there a reason we wouldn't want to always do this (like the configure command used to do)?

Reply:
The main reason for that is to keep `eas update` performance similar to how it was before. We can run most of these `ensureX` commands, but it will slow it down.

That being said, haven't benchmarked it.